core/src/main/java/hudson/security/ACL.java

```diff
@@ -69,7 +69,7 @@
      */
     public final void checkPermission(@NonNull Permission p) {
         Authentication a = Jenkins.getAuthentication2();
-        if (a.equals(SYSTEM2)) {
+        if (a.equals(SYSTEM2)) { // perhaps redundant given check in AccessControlled
```

Comment:
Seems cheap enough that we may as well keep this here just in case there are calls not routed through `AccessControlled`. TBD.